common/djangoapps/enrollment/data.py

```diff
@@ -104,6 +106,14 @@ def create_course_enrollment(username, course_id, mode, is_active):
         enrollment = get_course_enrollment(username, course_id)
         raise CourseEnrollmentExistsError(err.message, enrollment)
 
+    if enrollment.mode == 'credit' and is_credit_course(course_key):    # pylint: disable=no-member
+        credit_request = get_user_course_credit_request(username, course_key)
```

Comment:
Unfortunately, this isn't going to work.  Users won't be able to make a request for credit until they've paid, and the way we know that they've paid is by checking their enrollment and enrollment attributes.